test/main.cpp

```diff
@@ -368,16 +338,21 @@ EXPORT int main(int argc, char const* argv[])
 		if (filter && tests_to_run.count(_g_unit_tests[i].name) == 0)
 			continue;
 
-		std::snprintf(dir, sizeof(dir), "sub_%i", i);
-		std::string unit_dir = combine_path(current_working_directory(), dir);
+		std::string unit_dir = unit_dir_prefix + std::to_string(i);
 		error_code ec;
 		create_directory(unit_dir, ec);
 		if (ec)
 		{
-			std::printf("Failed to create unit directory: %s\n", ec.message().c_str());
+			std::printf("Failed to create unit test directory: %s\n", ec.message().c_str());
 			return 1;
 		}
-		if (!change_dir(dir)) return 1; // every unit test must have separate sandbox
+		change_directory(unit_dir, ec);
+		if (ec)
+		{
+			std::printf("Failed to change unit test directory: %s\n", ec.message().c_str());
+			return 1;
+		}
+		unit_dirs.push_back(unit_dir);
```

Comment:
I'd be happy to make this change after yours is in though. because I also don't understand why the directories are only cleaned up if ``TORRENT_LOGGING`` is not defined. that doesn't seem to make any sense